src/main/java/io/jenkins/plugins/appcenter/api/AppCenterServiceFactory.java

```diff
@@ -22,9 +32,13 @@
     private final Secret apiToken;
     private final String baseUrl;
 
-    public AppCenterServiceFactory(@Nonnull Secret apiToken, @Nullable URL baseUrl) {
+    private final ProxyConfiguration proxy;
+
+    public AppCenterServiceFactory(@Nonnull Secret apiToken, @Nullable URL baseUrl, @Nullable Jenkins jenkins) {
```

Comment:
I don't think we need access to the whole Jenkins instance in the factory. Only the Proxy configuration. So can we change this to `public AppCenterServiceFactory(@Nonnull Secret apiToken, @Nullable URL baseUrl, @Nullable ProxyConfiguration proxyConfiguration)` instead please.

Reply:
Also the docs for `Jenkins::getInstanceOrNull` says that we shouldn't be sending over the Jenkins object over remote channels. 

> For those cases you really should rewrite your code so that when the Callable is sent over the remoting channel it can do whatever it needs without ever referring to Jenkins for example, gather any information you need on the master side before constructing the callable.

Reply:
Good point. Changed.